app/models/pilot_flight_m/copy_to_flight.rb

```diff
@@ -0,0 +1,29 @@
+module PilotFlightM
+  module CopyToFlight
+    def copy_to_flight(flight)
+      new_attrs = self.attributes
+      new_attrs.delete('id')
+      new_attrs.delete('flight_id')
+      pf = flight.pilot_flights.find_by(pilot_id: self.pilot_id)
+      if (pf)
+        pf.update!(new_attrs)
+      else
+        pf = flight.pilot_flights.create!(new_attrs)
+      end
+      copy_grades_to(pf)
+      pf
+    end
+
+    def copy_grades_to(pf)
+      new_judges = self.scores.collect(&:judge_id)
```

Comment:
My previous comments about confusing variable names and `delete_all` / `destroy_all` also apply to this method.